zqd/handlers.go

```diff
@@ -112,6 +112,47 @@ func handleSearch(c *Core, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func handleWorker(c *Core, w http.ResponseWriter, httpReq *http.Request) {
+	var req api.WorkerRequest
+	if !request(c, w, httpReq, &req) {
+		return
+	}
+
+	space, err := c.spaces.Get(req.Space)
+	if err != nil {
+		respondError(c, w, httpReq, err)
+		return
+	}
+
+	ctx, cancel, err := space.StartOp(httpReq.Context())
+	if err != nil {
+		respondError(c, w, httpReq, err)
+		return
+	}
+	defer cancel()
+
+	srch, err := search.NewWorkerOp(req)
+	if err != nil {
+		// XXX This always returns bad request but should return status codes
+		// that reflect the nature of the returned error.
```

Comment:
I don't see any errors in `search.NewWorkerOp` that shouldn't be a 400 errors. Also, if there is an error with `ast.UnpackJSON` I believe the status would be 500 (the returned error is unwrapped).